rp2040-hal/src/timer.rs

```diff
@@ -14,10 +14,26 @@ use crate::atomic_register_access::{write_bitmask_clear, write_bitmask_set};
 use crate::pac::{RESETS, TIMER};
 use crate::resets::SubsystemReset;
 use core::marker::PhantomData;
+use core::sync::atomic::{AtomicU8, Ordering};
 
 /// Instant type used by the Timer & Alarm methods.
 pub type Instant = TimerInstantU64<1_000_000>;
 
+static ALARMS: AtomicU8 = AtomicU8::new(0x0F);
+fn take_alarm(mask: u8) -> bool {
+    critical_section::with(|_| {
+        let alarms = ALARMS.load(Ordering::Relaxed);
+        ALARMS.store(alarms & !mask, Ordering::Relaxed);
```

Comment:
I'm always unsure about this: Does `critical_section` provide any ordering guarantees?
The docs seem to be quiet about that topic. So maybe al the critical section guarantees is that no two threads can be inside a critical section at the same time.
If that's true, it may be necessary to use a stronger ordering here.
(At least formally, if we only consider rust guarantees. In practice, the rp2040 probably acts sufficiently benign that Relaxed will do...)

Reply:
the cortex-m0 does not reorder instructions or data accesses and our implementation (and cortex_m::interrupt::enable/disable) has compiler fences that should prevent bad things to happen.

Reply:
Yes, discussed that on the rust embedded matrix channel as well. Even though the docs don't spell it out, the critical section should guarantee ordering.